__tests__/app.test.js

```diff
@@ -173,3 +173,73 @@ describe("GET /api/articles/:article_id/comments", () => {
       });
   });
 });
+describe("POST /api/articles/:article_id/comments", () => {
+  test("post 201 update articles to include the new comment and return that comment", () => {
+    const comment = {
+      body: "Testing, testing, 123",
+      username: "icellusedkars",
+    };
+    return request(app)
+      .post("/api/articles/9/comments")
+      .send(comment)
+      .expect(201)
+      .then((response) => {
+        const { body } = response;
+        expect(body).toHaveProperty("comment_id");
+        expect(Object.keys(body).length).toBe(6);
+        expect(body).toMatchObject({
+          author: expect.any(String),
+          body: expect.any(String),
+          article_id: expect.any(Number),
+          votes: expect.any(Number),
+          created_at: expect.any(String),
+          comment_id: expect.any(Number),
+        });
+      });
+  });
+  test("STATUS 400 responds with a 400 and correct message for an invalid article id", () => {
+    const comment = {
+      body: "Testing, testing, 123",
+      username: "icellusedkars",
+    };
+
+    return request(app)
+      .post("/api/articles/hello/comments")
+      .send(comment)
+      .expect(400)
+      .then(({ body }) => {
+        const { msg } = body;
+        expect(msg).toBe("Invalid article id");
+      });
+  });
+  test("STATUS 404 responds with a 404 and correct message for a valid article that does not exist", () => {
+    const comment = {
+      body: "Testing, testing, 123",
+      username: "icellusedkars",
+    };
+
+    return request(app)
+      .post("/api/articles/999999/comments")
+      .send(comment)
+      .expect(404)
+      .then(({ body }) => {
+        const { msg } = body;
+        expect(msg).toBe("Username or article id does not exist");
+      });
+  });
+  test("STATUS 404 responds with a 404 and correct message for a valid article with username that does not exist", () => {
+    const comment = {
+      body: "Testing, testing, 123",
+      username: "broken",
+    };
+
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(comment)
+      .expect(404)
+      .then(({ body }) => {
+        const { msg } = body;
+        expect(msg).toBe("Username or article id does not exist");
+      });
+  });
+});
```

Comment:
Really nice testing so far! Could we just add the following two please...

- Status 201, ignores unnecessary properties (i.e. if we have an extra property as well as username/body)
- Status 400, missing required field(s), e.g. no username or body properties